backend/app/src/main/java/com/moh/phlat/backend/service/FileServiceImpl.java

```diff
@@ -250,56 +251,68 @@ private void copyInputSourceDataToProcessData(Long controlTableId, String authen
+		    processData.setSourceStatus(s.getSourceStatus()); 
+		    processData.setPcnClinicStatus(s.getPcnClinicStatus());
+		    processData.setHdsEffectiveStartDate(s.getHdsEffectiveStartDate());   
+		    processData.setFacAddressUnit(s.getFacAddressUnit()); 			    
+		    processData.setFacBuildingName(s.getFacBuildingName()); 	
 	        processData.setPhysicalAddr1(s.getPhysicalAddr1());
 	        processData.setPhysicalAddr2(s.getPhysicalAddr2());	  
 	        processData.setPhysicalAddr3(s.getPhysicalAddr3());
 	        processData.setPhysicalAddr4(s.getPhysicalAddr4());
 	        processData.setPhysicalCity(s.getPhysicalCity());        
 	        processData.setPhysicalProvince(s.getPhysicalProvince());      
 	        processData.setPhysicalPcode(s.getPhysicalPcode());      
-	        processData.setPhysicalCountry(s.getPhysicalCountry());     
-	        processData.setPhysAddrIsPrivate(s.getPhysAddrIsPrivate());    	        
+	        processData.setPhysicalCountry(s.getPhysicalCountry()); 
+	        processData.setPhysicalAddrPrpsTypeCd(s.getPhysicalAddrPrpsTypeCd()); 
 	        processData.setMailAddr1(s.getMailAddr1());
 	        processData.setMailAddr2(s.getMailAddr2());	        
 	        processData.setMailAddr3(s.getMailAddr3());
 	        processData.setMailAddr4(s.getMailAddr4());
 	        processData.setMailCity(s.getMailCity());        
 	        processData.setMailBc(s.getMailBc());      
 	        processData.setMailPcode(s.getMailPcode());      
-	        processData.setMailCountry(s.getMailCountry());     
-	        processData.setMailAddrIsPrivate(s.getMailAddrIsPrivate());  	        
+	        processData.setMailCountry(s.getMailCountry());
+	        processData.setMailAddressValidationStatus("");
 	        processData.setCreatedAt(s.getCreatedAt());
 	        processData.setCreatedBy(authenticateUserId);
 	        
-	        if (s.getDoNotLoad().equals("Y")) {
-		        processData.setRowstatusCode("DO_NOT_LOAD"); 
+	        // set default values
+		if (_control.getLoadTypeHds()) {
+		    processData.setHdsUserChid(_control.getBatchLabelName());
+		    processData.setHdsInvalidatedDts("9999-12-30");	        
+		    processData.setHdsEffectiveEndDate("9999-12-30");  		    
+		    processData.setHdsStatus("Active");   		    	
+	            processData.setHdsCategoryCode("ORGANIZATION");
+		    if (s.getHdsPreferredNameFlag().isEmpty()) {
+			processData.setHdsPreferredNameFlag("Y");   
+		    }	        
+		    if (s.getHdsType().isEmpty()) {
+			if (s.getStakeholder().equals("PHARMACY")) {
+			   processData.setHdsType("Pharmacy");
+		      } else if (s.getStakeholder().equals("CLINIC")) {
+		           processData.setHdsType("Clinic");
+		        }
+		    }
+	        }
+		    
+	        if (_control.getLoadTypeFacility()) {
+		    processData.setFacRelnType("LOCATION OF"); 		
+		    processData.setFacTypeCode("BUILDING"); 		
+	        }
+		    
+	        
+	        if (s.getDoNotLoadFlag().equals("Y")) {
```

Comment:
Everywhere, use constants first for comparison to avoid NPE

Reply:
This has been fixed.